qiskit_optimization/applications/ising/base_optimization_application.py

```diff
@@ -0,0 +1,36 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from abc import ABC, abstractmethod
+
+from qiskit_optimization.algorithms import OptimizationResult
+
+
+class BaseOptimizationApplication(ABC):
```

Comment:
I like `OptimizationApplication`. We can get rid of the `ising` folder. For `qiskit_finance` we should then also restructure to `applications.optimization` and `applications.estimation`. In case we have more problem classes than QPs at a later stage, we might want to introduce some more structure, but I agree, right now, the `ising` doesn't help much.